[quant][docs] README for FX Graph Mode Quantization

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85070

Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][docs] README for FX Graph Mode Quantization

Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][docs] README for FX Graph Mode Quantization"


Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/ao/quantization/fx/README.md:
```diff
@@ -0,0 +1,407 @@
+## Overview
+The FX graph representation is pretty close to python/eager mode, it preserves many python/eager mode constructs like modules, functionals, torch ops, so overall the implementation reuses some of building blocks and utilities from eager mode quantization, this includes the QConfig, QConfig propagation (might be removed), fused modules, QAT module, quantized modules, QAT module swapping utility. A ...
+
+### Quantization Flow Code Example
```
should we delete this, since it's covered by user docs?

jerryzh168 replied:
sure, sounds good

vkuzo commented on torch/ao/quantization/fx/README.md:
```diff
@@ -0,0 +1,407 @@
+
+## High Level Flow with Simple Example
+
+Floating Point Model — (1.1 `_fuse_fx`) —> Fused Model —-- (1.2 QAT Module Swap) —--> Model with QAT modules —-- (1.3 Insert Observers) —---> Prepared Model
```
nit: make multiline, the line is hard to read right now?

vkuzo commented on torch/ao/quantization/fx/README.md:
````diff
@@ -0,0 +1,407 @@
+    return linear
+```
+
+In eager mode quantization, users need to call `fuse_modules` or `fuse_modules_qat` and provide a list of fully qualified names for the submodules that need to be fused. We will get the same result as the eager mode fusion.
````
Would it make sense to remove the comparisons with eager mode from this doc, and just explain what's happening?  This way people who don't understand eager mode would be able to follow better.

vkuzo reviewed (approved):
thanks for writing this!  Looks great, we can iterate on this in future PRs.

jerryzh168 committed:
Update on "[quant][docs] README for FX Graph Mode Quantization"


Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][docs] README for FX Graph Mode Quantization"


Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][docs] README for FX Graph Mode Quantization"


Summary:
This is a developer-oriented design doc/README for FX Graph Mode Quantization, the goal for the doc is for new developers of
FX Graph Mode Quantization to get familiarized with the high level algorithm of FX Graph Mode Quantization and ramp up quickly

Test Plan:
no test needed

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge

jerryzh168 commented:
@pytorchbot merge
